### PR TITLE
Create Interop Dockers for running besu against GoQuorum

### PR DESCRIPTION
**Pull request by rain-on**

### CODE REVIEW

File: docker_interop/env.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+export BESU_IMAGE="hyperledger/besu:21.1.3-SNAPSHOT-openjdk-11"
+export QUORUM_IMAGE="localquorum:1.0"
```
Comment (usmansaleem) [inline docker_interop/env.sh:4]: localquorum suggests it is built locally?

**Review by usmansaleem — APPROVED** [review]
LGTM - minor question around GoQuorum docker image.

File: docker_interop/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        --gasprice 0
+    networks:
+      app_net:
+        ipv4_address: 172.16.239.15
+    restart: "no"
+    volumes:
+      - ./:/importData
+    environment:
+      - PRIVATE_CONFIG=ignore
+  besu-node-0:
```
Comment (jframe) [inline docker_interop/docker-compose.yml:206]: be good to have a few more besu nodes on the network instead just the one

File: docker_interop/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+This docker-composer allows a network of GoQuorum and Besu Ethereum nodes to be run up
+in a network.
+
+This network can then be queried via Json RPC (or by log parsing) to determine if the nodes
+have peered and are producing blocks.
+
+## Operation
+To run this network:
+0. Ensure a Docker server is running on the local PC
+1. Source the env.sh script, to set the tag of the Besu and GoQuorum images to be used (not it must do so in THIS environment, not a sub-shell, thus 'source' not execute)
```
Comment (jframe) [inline docker_interop/README.md:13]: maybe include the commands for building the docker images in besu and goquorum

File: docker_interop/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+## Potential Issues
+The node-keys used by the nodes _must_ be hardcoded (rather than randomly assigned) to ensure the nodes
+which are QBFT validators align with the content of the Genesis File's ExtraData field (which encodes the
+addresses of the nodes which are validators).
+
+Thus, there is some manual effort required if you wish to reconfigure the network (eg bring in an extra besu validator).
+
+The steps to do this:
+1. Create new SECP256k1 keypair for the node
+2. Create a new 'ExtraData' string, and insert into the Genesis file
```
Comment (jframe) [inline docker_interop/README.md:31]: how is the extradata string created?

File: docker_interop/besu_genesis.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{
+	"config": {
+		"chainId": 2017,
+		"homesteadBlock": 0,
+		"qbft": {
+			"epoch": 30000,
+			"policy": 0
```
Comment (jframe) [inline docker_interop/besu_genesis.json:7]: what's policy for?

File: docker_interop/README.md (hunk 4 above)
Comment (rain-on) [reply docker_interop/README.md:31]: its just hard-coded - there's only 1 genesis file ... but want to draw attention to the fact this is hard!

File: docker_interop/besu_genesis.json (hunk 5 above)
Comment (rain-on) [reply docker_interop/besu_genesis.json:7]: bleugh - hang-over, its really about "do we change proposer every block, or only on faiure"? We don't use it - but quoruom does

**Comment by usmansaleem** [comment]
Picked up in #9

**Pull request closed without merging**